service/rpc/util.go

```diff
@@ -0,0 +1,16 @@
+package rpc
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func writeError(w http.ResponseWriter, statusCode int, endpoint string, err error) {
+	w.WriteHeader(statusCode)
+	errBody, _ := json.Marshal(err.Error())
```

Comment:
Why not check for err here?

Reply:
I assumed that we can be pretty sure about marshaling a string. I can add this.

Reply:
Yeah let's add the err check too just in case. Otherwise good to go